### ISSUE TITLE
Questions about network with jpeg

### ISSUE THREAD

**Issue by brighteast** [pkuxmq/Invertible-Image-Rescaling#42]
Thanks for your great work.
I read your paper really interestingly.
And I have a question.
I wonder how to train 'IRN_model_CRM'. I guess first you train IRN, afterwards you train CRM('only_jpeg_reconstruction')... is that right?
I want to know exactly how to train this network.
Thank you :)

**Comment by pkuxmq** [comment]
Yes, in our experiments we directly use the pretrained IRN model and only train the CRM module with 'only_jpeg_reconstruction' and only LR reconstruction loss. It is also possible to train CRM with the joint loss of HR reconstruction (backpropagated through the IRN model) or finetune IRN model at the same time. We added these choices in the code (e.g. in the 'train_IRN-Compression_x2_q90.yml' file, setting 'add_joint_loss' and 'lambda_joint_back' can add the joint loss, and setting 'only_jpeg_reconstruction' as False will also finetune IRN), but currently we did not conduct experiments to verify this.

**Comment by brighteast** [comment]
Oh really thankful for your quick reply. Then in detail you first train three loss (l_forw_fit + l_forw_ce + l_back_rec), and then train four loss (l_forw_fit + l_back_rec + l_back_gan + l_back_fea), lastly finetune CRM using 'l_jpeg_rec'.. Is it correct?

Actually I'm confused about Gan loss. When I finetune model using 'only_jpeg_reconstruction = False', I wonder why you use 'l_forw_ce' instead of Gan loss
Thank you!!

**Comment by pkuxmq** [comment]
In the experiments, we use the IRN model with three losses (l_forw_fit + l_forw_ce + l_back_rec) (i.e. the pretraining stage as mentioned in the paper) rather than IRN+ with GAN loss and feature loss. Because IRN+ encourages more perceptual quality than pixel similarity, so it has higher perceptual metrics but lower PSNR/SSIM, and we use IRN for higher PSNR metrics. So in detail we first train IRN with three losses and then train CRM using 'l_jpeg_rec'. It is also possible to use the IRN+ model for better perceptual quality. Current codes focus on IRN and could be modified for IRN+.

**Comment by brighteast** [comment]
I totally understand. Really thanks for your kindness. Have a nice day!
